tests/h/auth/policy_test.py

```diff
@@ -124,38 +125,74 @@ def test_unauthenticated_userid_is_none_if_no_token(self, pyramid_request):
 
         assert policy.unauthenticated_userid(pyramid_request) is None
 
-    def test_unauthenticated_userid_returns_userid_from_token(self, fake_token, pyramid_request):
+    def test_unauthenticated_userid_returns_userid_from_token(self, pyramid_request):
         policy = TokenAuthenticationPolicy()
-        pyramid_request.auth_token = fake_token
+        pyramid_request.auth_token = 'valid123'
 
         result = policy.unauthenticated_userid(pyramid_request)
 
         assert result == 'acct:foo@example.com'
 
-    def test_unauthenticated_userid_returns_none_if_token_invalid(self, pyramid_request):
+    def test_unauthenticated_userid_returns_none_if_token_invalid(self, pyramid_request, token_service):
         policy = TokenAuthenticationPolicy()
-        token = DummyToken(valid=False)
-        pyramid_request.auth_token = token
+        token_service.validate.return_value = None
+        pyramid_request.auth_token = 'abcd123'
 
         result = policy.unauthenticated_userid(pyramid_request)
 
         assert result is None
 
-    def test_authenticated_userid_uses_callback(self, fake_token, pyramid_request):
+    def test_unauthenticated_userid_returns_userid_from_query_params_token(self, pyramid_request):
+        """When the path is `/ws` then we look into the query string parameters as well."""
+
+        policy = TokenAuthenticationPolicy()
+        pyramid_request.GET['access_token'] = 'valid123'
+        pyramid_request.path = '/ws'
```

Comment:
An alternative to checking a hard-coded path might be to check one of the headers in the [upgrade request](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_servers#Client_Handshake_Request) such as the `Sec-WebSocket-Version` header, which browser-based clients are not able to set themselves. This would work if WS connections were ever served from a different path. Thoughts?